experiment/secret-sync-controller/secret_sync_spec.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"context"
+	// b64 "encoding/base64"
+	// "bytes"
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"strconv"
+	"strings"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	secretmanagerpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type SecretSyncSpec struct {
+	// the target can be either a K8s secret or a SecretManager secret
+	Source      Target `yaml:"source"`
+	Destination Target `yaml:"destination"`
+}
+
+type Target struct {
+	// assert that one of the two should be `nil`
+	Kubernetes    *KubernetesSpec    `yaml:"kubernetes"`
```

Comment:
Allow for these fields to not exist
```suggestion
	Kubernetes    *KubernetesSpec    `yaml:"kubernetes,omitempty"`
```

Reply:
Add 'omitempty' tags to 
- Secret
- DenyList
- Kubernetes
- SecretManager
fields